compiler/src/Language/Bond/Codegen/Java/Class_java.hs

```diff
@@ -10,25 +10,216 @@ module Language.Bond.Codegen.Java.Class_java
     ) where
 
 import Prelude
-import Data.Monoid
 import Data.Text.Lazy (Text)
 import Text.Shakespeare.Text
 import Language.Bond.Syntax.Types
 import Language.Bond.Util
 import Language.Bond.Codegen.TypeMapping
 import Language.Bond.Codegen.Util
-import Language.Bond.Codegen.Java.Schema
-import Language.Bond.Codegen.Java.SerializationMethods
-import qualified Language.Bond.Codegen.Java.Util as Java
+import Language.Bond.Codegen.Java.Util
+
 
 -- field -> public field
 data JavaFieldMapping = JavaPublicFields deriving Eq
 
+
+-- returns the name of the struct field descriptor type (a protected nested class within StructBondType)
+structFieldDescriptorTypeName :: MappingContext -> Type -> Text
+structFieldDescriptorTypeName java type_ = typeName type_
+    where
+        typeName (BT_Maybe BT_Int8) = [lt|com.microsoft.bond.StructBondType.SomethingInt8StructField|]
+        typeName BT_Int8 = [lt|com.microsoft.bond.StructBondType.Int8StructField|]
```

Comment:
This way, all types are together. As we discussed this morning, let's get the codegen changes in and then address formatting of Haskell code later.

---
In reply to: [124937143](https://github.com/Microsoft/bond/pull/517#discussion_r124937143) [](ancestors = 124937143)